### PR TITLE
[Photos] Correct nullability attributes for the Photos framework

### PR DESCRIPTION
**Pull request by rpendleton**
I corrected the nullability attributes of the Photos bindings based on the instructions from @spouliot in [#6154](https://github.com/xamarin/xamarin-macios/issues/6154#issuecomment-620014843).

### CODE REVIEW

**Comment by dalexsoto** [comment]
wow thanks a bunch!!

**Comment by dalexsoto** [comment]
build

File: src/photos.cs [hunk 1]
```diff
@@ -400,9 +412,11 @@ interface PHAssetResourceRequestOptions : NSCopying
 	interface PHChange {
 
 		[Export ("changeDetailsForObject:")]
-		PHObjectChangeDetails GetObjectChangeDetails ([NullAllowed] PHObject obj);
```
Comment (rpendleton) [inline src/photos.cs:403]: Is removing this `[NullAllowed]` attribute from the first method parameter a breaking change?

If I understand things correctly, this change will result in code being generated that throws a `NullArgumentException` when given null, and that wouldn't have happened until now. I tried to test passing `null` to this method before my change and didn't get an exception from Apple, but I'm also not sure if that's just relying on undocumented behavior.

Comment (spouliot) [reply src/photos.cs:403]: > Is removing this [NullAllowed] attribute from the first method parameter a breaking change?

It's a functional change but not a **breaking change** (as defined by .net). Nullability information is for
* macios generator (bgen) which add null-check **inside** methods; and
* csc (roslyn) compiler (for warnings)

This means binaries will still run, source will still build (might warn or error if `/warnaserror` is used, but that's fine).

> not sure if that's just relying on undocumented behavior.

The nullability information has not always been available from Apple. The old attributes were added based on headers comments, documentation, samples, unit tests and common sense.

**Review by spouliot — APPROVED** [review]
Looks identical to what I would have done 😄 
Thanks a lot!

**Review by rolfbjarne — APPROVED** [review]
This looks great! Thanks!

**Review by whitneyschmidt — APPROVED** [review]

**Review by mandel-macaque — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Comment by spouliot** [comment]
@monojenkins backport to xcode12.2

**Pull request merged**